### PR TITLE
Continually retries the first connection rather than exiting on failure

### PR DESCRIPTION
**Pull request by leonsodhi**

### CODE REVIEW

**Comment by leonsodhi** [comment]
This'll need more testing once #97 is live.

File: syslog/syslog.go [hunk 1]
```diff
@@ -90,25 +90,20 @@ type Logger struct {
...
+	logger := &Logger{
+		ClientHostname: clientHostname,
+		network:        network,
+		raddr:          raddr,
+		rootCAs:        rootCAs,
+		Packets:        make(chan Packet, 100),
+		Errors:         make(chan error, 0),
 	}
+
+	logger.connect()
```
Comment (eric) [inline syslog/syslog.go]: This is going to block indefinitely until a successful connection is made, which means that [this logging](https://github.com/papertrail/remote_syslog2/blob/retry-first-connection/remote_syslog.go#L120) will never happen for errors during the initial connection.

I would suggest moving this back to a single `dial()` call above creating the `Logger` instance which will give one chance to connect, but then return an instance of the `Logger` which will allow the debug log to report what is going on.

Comment (leonsodhi) [inline syslog/syslog.go]: Ah, yes, I hadn't thought of that. The problem is if I go back to a single `dial()` and return it, there's nothing to make it retry and we'd have to wait until a log message is sent to see any other errors.

I wonder if an `init` method could be added to create an Error channel before `Dial()` is called in `remote_syslog.go`, and then a goroutine could run:

```
for err := range logger.Errors {
 		log.Errorf("Syslog error: %v", err)
 	}
```

What do you think?

Comment (eric) [inline syslog/syslog.go]: Why do we care about retrying forever if there isn't anything to send? What is the benefit of ensuring that a connection is always established if there isn't something that needs to be sent?

**Pull request closed without merging**